v2/codetags/types.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package codetags
+
+import (
+	"fmt"
+	"strings"
+)
+
+// TypedTag represents a single comment tag with typed args.
+type TypedTag struct {
+	// Name is the name of the tag with no arguments.
+	Name string
+	// Args is a list of optional arguments to the tag.
+	Args []Arg
+	// Value is the string representation of scalar tag values.
+	// Only valid when ValueType is ValueTypeString, ValueTypeBool or ValueTypeInt.
+	Value string
+
+	// ValueTag is value of the tag when the tag's value is another tag.
+	// Only valid when ValueType is ValueTypeTag.
+	ValueTag *TypedTag
+
+	// ValueType is the type of the value.
+	// If ValueType is ValueTypeTag, ValueTag is set.
+	// If ValueType is ValueTypeString, ValueTypeBool or ValueTypeInt, Value is the string representation of the value.
+	// If ValueType is empty, the tag has no value.
+	ValueType ValueType
```

Comment:
Alternative's I've considered:

- Leave chained tags as string values in `Value` instead of providing a field to access the parsed tag.
- Change `Parse()` to return a slice of `TypedTag` where each item in the slice represents a chained tag.
- Change `Value` to be either `any` or an interface that can either be a `string` or a `TypedTag`.